### PR TITLE
project (data/resource) and building block (data)

### PR DESCRIPTION
**Pull request by henryde**

### CODE REVIEW

File: examples/provider-install-verification/main.tf [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      source = "meshcloud/meshstack"
+    }
+  }
+}
+
+locals {
+  workspace = "henrys-workspace"
+}
+
+provider "meshstack" {
```
Comment (j0g3sc) [inline examples/provider-install-verification/main.tf:13]: probably not critical, but better put secrets here.
guess the whole example can be reworked a bit later.

File: internal/provider/buildingblock.go [hunk 2]
```diff
@@ -1,43 +1,42 @@
...
+	Inputs               []MeshBuildingBlockIO     `json:"inputs" tfsdk:"inputs"`
+	ParentBuildingBlocks []MeshBuildingBlockParent `json:"parentBuildingBlocks" tfsdk:"parent_building_blocks"`
 }
 
 type MeshBuildingBlockIO struct {
-	Key       string `json:"key"`
-	Value     any    `json:"value"`
-	ValueType string `json:"valueType"`
+	Key       string `json:"key" tfsdk:"key"`
+	Value     string `json:"value" tfsdk:"value"`
```
Comment (j0g3sc) [inline internal/provider/buildingblock.go:30]: we support int and bool as well in the backend.

File: internal/provider/client.go [hunk 3]
```diff
@@ -141,50 +154,213 @@ func (c *MeshStackProviderClient) lookUpEndpoints() error {
...
+	}
+
+	defer res.Body.Close()
+
+	data, err := io.ReadAll(res.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: create will also succeed if the project already exists
```
Comment (j0g3sc) [inline internal/provider/client.go:286]: shouldn't be the case anymore

**Review by j0g3sc — APPROVED** [review]
minor comments, mainly validated that the schemata look correct to me :+1:

File: internal/provider/buildingblock.go (hunk 2 above)
Comment (henryde) [reply internal/provider/buildingblock.go]: Hm. according to the docs we also have FILE, LIST, SINGLE_SELECT.

**Pull request merged**